sdks/java/core/src/main/java/org/apache/beam/sdk/io/LocalResourceId.java

```diff
@@ -35,7 +35,8 @@
  */
 class LocalResourceId implements ResourceId {
 
-  private final Path path;
+  private final String pathString;
+  private transient volatile Path cachedPath;
```

Comment:
I am trying to understand the cachedPath.
Is it for relative path string, such as "./output"?

So, LocalResourceId could change after de-serialization?